src/lib/ListMetrics.js

```diff
@@ -1,35 +1,73 @@
 import async from 'async';
 import { errors } from 'arsenal';
 import { getMetricFromKey, getKeys, generateStateKey } from './schema';
+import metricResponseJSON from '../../models/metricResponse';
 
 /**
-* Provides static methods to get bucket level metrics
+* Provides methods to get metrics of different levels
 */
-export default class Buckets {
+export default class ListMetrics {
 
     /**
-    * Callback for getting metrics for a list of buckets
-    * @callback Buckets~bucketsMetricsCb
+     * Assign the metric property to an instance of this class
+     * @param {string} metric - The metric type (e.g., 'buckets', 'accounts')
+     */
+    constructor(metric) {
+        this.metric = metric;
+    }
+
+    /**
+     * Create the metric object to retrieve data from schema methods
+     * @param {string} resource - The resource to get metrics for
+     * @return {object} obj - Object with a key-value pair for a schema method
+     */
+    _getSchemaObject(resource) {
+        let type;
+        if (this.metric === 'buckets') {
+            type = 'bucket';
+        } else if (this.metric === 'accounts') {
+            type = 'accountId';
+        }
+        const obj = {};
+        obj[type] = resource;
+        return obj;
+    }
+
+    // Create the metric response object for a given metric.
+    _getMetricResponse(resource, start, end) {
+        const metricResponse = Object.assign({}, metricResponseJSON);
+        metricResponse.timeRange = [start, end];
+        const map = {
+            buckets: 'bucketName',
+            accounts: 'accountId',
+        };
+        metricResponse[map[this.metric]] = resource;
```

Comment:
I'm curious what 'map' is doing here. Wouldn't it be enough to just say metricResponse[this.metric] = resource?